skimage/measure/_marching_cubes_lewiner_cy.pyx

```diff
@@ -190,6 +187,13 @@ cdef class Cell:
         # Allocate face layers
         self.faceLayer1 = <int *>malloc(self.nx*self.ny*4 * sizeof(int))
         self.faceLayer2 = <int *>malloc(self.nx*self.ny*4 * sizeof(int))
+
+        if (self.faceLayer1 is NULL or self.faceLayer2 is NULL or
+            self.vv is NULL or self.vg is NULL or self._vertices is NULL or
+            self._normals is NULL or self._values is NULL or
+            self._faces is NULL):
+            raise MemoryError()
+
```

Comment:
What happens here if we do not de-allocate the values that already succeeded?

Reply:
The should be released in `__dealloc__`, no?

Reply:
is __dealloc__ safe to any failed memory allocations? seems like they all need to be guarded right?

Reply:
Yes, `free` is a no-op with NULL pointers, and will work correctly with allocated pointers. It is *not* safe to `free` a pointer that has already been freed. However that should not happen in this case, I believe.